### PR TITLE
Migrate form types to FQCN / autowire

### PR DESCRIPTION
**Pull request by FabienPapet**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Refactor form types service definitions
| Type?             | improvement
| Category?         | CO
| BC breaks?        | yes
| Deprecations?     |  no
| How to test ?     |  This is a major refactoring on how forms are generated. It should have no impact at all, but it requires to go to every form in the application to check. A lot of checks can be done with automated tests, as we only check if the form is displayed. If the form is displayed, it will be working as it used to. The acceptance here is just the form to be successfully displayed.

### CODE REVIEW

**Comment by FabienPapet** [comment]
https://github.com/FabienPapet/ga.tests.ui.pr/actions/runs/4064714112

File: src/PrestaShopBundle/Resources/config/services/bundle/form/form_type.yml
Comment (lartist) [inline src/PrestaShopBundle/Resources/config/services/bundle/form/form_type.yml:38]: Should we use generic binding instead, to mutualize the definition:
```yaml
services:
  _defaults:
    public: false
    autowire: true
    autoconfigure: true
    bind:
        $saveFormLocaleChoice: "@=service('prestashop.adapter.legacy.configuration').getBoolean('PS_BO_ALLOW_EMPLOYEE_FORM_LANG')"
        ...
```

What do you think?

Comment (FabienPapet) [reply src/PrestaShopBundle/Resources/config/services/bundle/form/form_type.yml:38]: That makes sense, but for the existing services definitions, the same attribute can have many names, sometimes we can have `currency`, `currencyIsoCode` or `defaultCurrencyIsoCode` for the same parameter inside a constructor on two services.

I agree that this is weird and this should be avoided, but we can have some side effects  renaming them. And that would make this PR way too complicated to review and maintain. This can be done in a second time.

**Review by mflasquin — COMMENTED** [review]
In the PR description It's indicate "BC break : no" can you change to "yes" ?
Can you relaunch tests : https://github.com/FabienPapet/ga.tests.ui.pr/actions/runs/4064714112

File: src/PrestaShopBundle/Resources/config/services/bundle/form/form_type.yml
Comment (lartist) [reply src/PrestaShopBundle/Resources/config/services/bundle/form/form_type.yml:38]: Ok for me. Plz consider the following message from @mflasquin and I'll approve 👍

File: src/PrestaShopBundle/Resources/config/services/bundle/form/form_type.yml
Comment (lartist) [inline src/PrestaShopBundle/Resources/config/services/bundle/form/form_type.yml]: Why do you completly remove this argument

**Review by lartist — DISMISSED** [review]

**Review by mflasquin — DISMISSED** [review]

**Review by mflasquin — DISMISSED** [review]

**Review by lartist — DISMISSED** [review]

**Review by HanaRebaiQA — COMMENTED** [review]
Hello @FabienPapet 

I have checked your PR. I found these issues : 

-**Issue 1 :** The automated tests are failed in these tests : **BO:Orders** and **BO:Shipping**
For the **BO:Catalog** and **FO**, they are already failed on develop and without PR.

Here is the test run :
https://github.com/HanaRebaiQA/testing_pr/actions/runs/4112094347

Could you please check if its related to your PR ?
ping @PrestaShop/qa-automation What do you think?

-**Issue 2 :** A 500 error on the the `Payement>Preferences` page 

image


https://user-images.githubusercontent.com/92912932/217226981-0ad3e722-7b03-481e-a4d7-a4932ce267b0.mp4

Could you please check it ?

Thank You!

**Comment by FabienPapet** [comment]
Page is working fine on my computer, the problem may be on you machine, 
image

**Comment by nesrineabdmouleh** [comment]
Hello @FabienPapet

Thank you @HanaRebaiQA for launching the UI tests :) 

The first issue in '**BO>shipping**' that the list of carriers in **'BO/Shipping/Preferences'** is changed

In the PR
Capture d’écran de 2023-02-07 14-55-12

In develop
Capture d’écran de 2023-02-07 14-57-29

The second issue in '**BO>orders**' that the list of products in quick view order page is changed!
In the PR 
Capture d’écran de 2023-02-07 15-30-22

in Develop
Capture d’écran de 2023-02-07 15-30-13

I launched the same test in develop and in the PR => The order contains 11 products, in develop there are 11 but in the PR there are 8 (last product quantity = 4)

Could you please check it ?
Thank You!

**Comment by nesrineabdmouleh** [comment]
Hello @FabienPapet 

I just found that he error is not related to the PR, the failed tests of orders exist in Nightly => https://github.com/PrestaShop/PrestaShop/actions/runs/4109601320/jobs/7091611767#step:6:3190

**Comment by FabienPapet** [comment]
Fixed

**Review by mflasquin — DISMISSED** [review]

**Comment by HanaRebaiQA** [comment]
Hello @FabienPapet 

We need a second review approval before starting testing this PR.

Thank You!

**Review by lartist — DISMISSED** [review]

**Comment by AureRita** [comment]
Hi @FabienPapet 

Thank you for your correction, I tested your PR and I find this issue on Design > Email Theme : 

with your PR : 
image

without your PR : 
image

We can't, currently, check the modern theme, as you can see : 

[Untitled_ Feb 14, 2023 4_37 PM.webm](https://user-images.githubusercontent.com/100699445/218785524-a65d5487-1a0c-4476-a80c-b68962e2b039.webm)

Waiting for your feedback

**Review by atomiix — APPROVED** [review]

**Review by mflasquin — APPROVED** [review]

**Review by AureRita — APPROVED** [review]
Thank you for your modification, all works as planed 

Thank you

**Pull request merged**
